bundles/org.openhab.binding.sagercaster/src/main/java/org/openhab/binding/sagercaster/internal/handler/SagerCasterHandler.java

```diff
@@ -0,0 +1,239 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.sagercaster.internal.handler;
+
+import static org.eclipse.smarthome.core.library.unit.MetricPrefix.HECTO;
+import static org.openhab.binding.sagercaster.internal.SagerCasterBindingConstants.*;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import javax.measure.quantity.Angle;
+import javax.measure.quantity.Dimensionless;
+import javax.measure.quantity.Pressure;
+import javax.measure.quantity.Temperature;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.QuantityType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.library.unit.SIUnits;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.StateOption;
+import org.openhab.binding.sagercaster.internal.SagerWeatherCaster;
+import org.openhab.binding.sagercaster.internal.WindDirectionStateDescriptionProvider;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link SagerCasterHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Gaël L'hopital - Initial contribution
+ */
+@NonNullByDefault
+public class SagerCasterHandler extends BaseThingHandler {
+    private final static StringType FORECAST_PENDING = new StringType("0");
+    private final Logger logger = LoggerFactory.getLogger(SagerCasterHandler.class);
+    private final static SagerWeatherCaster sagerWeatherCaster = new SagerWeatherCaster();
```

Comment:
In an OSGI environment it discouraged to statically initialize classes within your bundle. Statically initializing makes garbage collection difficult and can cause issues when reloading the bundle. Instead you should make SagerWeatherCaster a osgi Component, like WindDirectionStateDescriptionProvider, and then have it passed into this class through the constructor, just like WindDirectionStateDescriptionProvider.

In addition turning SagerWeatherCaster into a Component would allow you to put the properties file loading into a separate method annotated with `@Activate` which could throw errors if property loading fails for whatever reason. An error would prevent the bundle from loading which I think would be a good think since you would rather have the bundle fail to load than for the bundle to operate improperly.